From: Simon McVittie <smcv@collabora.com>
Date: Thu, 31 Aug 2017 13:41:35 +0100
Subject: Put FLATPAK_SYSTEM_CACHE_DIR alongside other test data

This makes it easier to reason about. In the absence of
FLATPAK_SYSTEM_CACHE_DIR, it would use /var/tmp even during tests that
prefer /tmp.

Signed-off-by: Simon McVittie <smcv@collabora.com>
Forwarded: https://github.com/flatpak/flatpak/pull/988
---
 tests/libtest.sh    | 1 +
 tests/testlibrary.c | 6 ++++++
 2 files changed, 7 insertions(+)

diff --git a/tests/libtest.sh b/tests/libtest.sh
index 96f8b2a..5ab17bb 100644
--- a/tests/libtest.sh
+++ b/tests/libtest.sh
@@ -78,6 +78,7 @@ mkdir -p ${TEST_DATA_DIR}/home
 mkdir -p ${TEST_DATA_DIR}/runtime
 mkdir -p ${TEST_DATA_DIR}/system
 export FLATPAK_SYSTEM_DIR=${TEST_DATA_DIR}/system
+export FLATPAK_SYSTEM_CACHE_DIR=${TEST_DATA_DIR}/system-cache
 export FLATPAK_SYSTEM_HELPER_ON_SESSION=1
 
 export HOME=${TEST_DATA_DIR}/home
diff --git a/tests/testlibrary.c b/tests/testlibrary.c
index 9ad0ac1..7f3eae8 100644
--- a/tests/testlibrary.c
+++ b/tests/testlibrary.c
@@ -12,6 +12,7 @@
 static char *testdir;
 static char *flatpak_runtimedir;
 static char *flatpak_systemdir;
+static char *flatpak_systemcachedir;
 static char *flatpak_configdir;
 static char *flatpak_installationsdir;
 static char *gpg_homedir;
@@ -831,6 +832,11 @@ global_setup (void)
   g_setenv ("FLATPAK_SYSTEM_DIR", flatpak_systemdir, TRUE);
   g_test_message ("setting FLATPAK_SYSTEM_DIR=%s", flatpak_systemdir);
 
+  flatpak_systemcachedir = g_strconcat (testdir, "/system-cache", NULL);
+  g_mkdir_with_parents (flatpak_systemcachedir, S_IRWXU|S_IRWXG|S_IRWXO);
+  g_setenv ("FLATPAK_SYSTEM_CACHE_DIR", flatpak_systemcachedir, TRUE);
+  g_test_message ("setting FLATPAK_SYSTEM_CACHE_DIR=%s", flatpak_systemcachedir);
+
   flatpak_configdir = g_strconcat (testdir, "/config", NULL);
   g_mkdir_with_parents (flatpak_configdir, S_IRWXU|S_IRWXG|S_IRWXO);
   g_setenv ("FLATPAK_CONFIG_DIR", flatpak_configdir, TRUE);
